From c8e635088ddb8cbf059e96ed8dced0ab22cc24d3 Mon Sep 17 00:00:00 2001
From: Ding Wei <leo.ding@rock-chips.com>
Date: Sun, 21 Feb 2021 15:14:27 +0800
Subject: [PATCH] video: rockchip: mpp: issue for power off

when have task in running_list, it should be power off later.

Change-Id: I6df2d8ed11a05eba5753d78800c2175fc99ad0f1
Signed-off-by: Ding Wei <leo.ding@rock-chips.com>
---
 drivers/video/rockchip/mpp/mpp_common.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/mpp/mpp_common.c b/drivers/video/rockchip/mpp/mpp_common.c
index 87b48c60fe97..ec25e33a56e1 100644
--- a/drivers/video/rockchip/mpp/mpp_common.c
+++ b/drivers/video/rockchip/mpp/mpp_common.c
@@ -212,7 +212,8 @@ int mpp_power_off(struct mpp_dev *mpp)
 		mpp->hw_ops->clk_off(mpp);
 
 	pm_relax(mpp->dev);
-	if (mpp_taskqueue_get_pending_task(mpp->queue)) {
+	if (mpp_taskqueue_get_pending_task(mpp->queue) ||
+	    mpp_taskqueue_get_running_task(mpp->queue)) {
 		pm_runtime_mark_last_busy(mpp->dev);
 		pm_runtime_put_autosuspend(mpp->dev);
 	} else {
-- 
2.35.3

